ArduPlane/tailsitter.cpp

```diff
@@ -56,79 +151,75 @@ void QuadPlane::tailsitter_output(void)
+
+            // override AP_MotorsTailsitter throttles during back transition
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, throttle);
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttleLeft, throttle);
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttleRight, throttle);
+        }
+        if (tailsitter.motor_mask != 0) {
+            // set AP_MotorsMatrix throttles enabled for forward flight
+            motors->output_motor_mask(throttle * 0.01f, tailsitter.motor_mask, plane.rudder_dt);
         }
         return;
     }
 
-    // handle VTOL modes
-    // the MultiCopter rate controller has already been run in an earlier call 
-    // to motors_output() from quadplane.update()
-    motors_output(false);
+    if (assisted_flight && tailsitter_transition_fw_complete()) {
+        hold_stabilize(throttle * 0.01f);
+        motors_output(true);
+    } else {
+        motors_output(false);
+    }
+
+    // if in Q assist still a good idea to use copter I term and zero plane I to prevent windup
     plane.pitchController.reset_I();
     plane.rollController.reset_I();
 
     // pull in copter control outputs
-    SRV_Channels::set_output_scaled(SRV_Channel::k_aileron, (motors->get_yaw())*-SERVO_MAX);
-    SRV_Channels::set_output_scaled(SRV_Channel::k_elevator, (motors->get_pitch())*SERVO_MAX);
-    SRV_Channels::set_output_scaled(SRV_Channel::k_rudder, (motors->get_roll())*SERVO_MAX);
-    SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, (motors->get_throttle()) * 100);
-
-    if (hal.util->get_soft_armed()) {
-        // scale surfaces for throttle
-        tailsitter_speed_scaling();
-    }
+    float aileron = motors->get_yaw()*-SERVO_MAX;
+    float elevator = motors->get_pitch()*SERVO_MAX;
+    float rudder = motors->get_roll()*SERVO_MAX;
+    throttle = motors->get_throttle() * 100;
+    float tilt_left = SRV_Channels::get_output_scaled(SRV_Channel::k_tiltMotorLeft);
+    float tilt_right = SRV_Channels::get_output_scaled(SRV_Channel::k_tiltMotorRight);
 
     if (tailsitter.vectored_hover_gain > 0) {
-        // thrust vectoring VTOL modes
-        tilt_left = SRV_Channels::get_output_scaled(SRV_Channel::k_tiltMotorLeft);
-        tilt_right = SRV_Channels::get_output_scaled(SRV_Channel::k_tiltMotorRight);
         /*
           apply extra elevator when at high pitch errors, using a
           power law. This allows the motors to point straight up for
           takeoff without integrator windup
          */
-        int32_t pitch_error_cd = (plane.nav_pitch_cd - ahrs_view->pitch_sensor) * 0.5;
+        float des_pitch_cd = attitude_control->get_att_target_euler_cd().y;
+        int32_t pitch_error_cd = (des_pitch_cd - ahrs_view->pitch_sensor) * 0.5;
         float extra_pitch = constrain_float(pitch_error_cd, -SERVO_MAX, SERVO_MAX) / SERVO_MAX;
         float extra_sign = extra_pitch > 0?1:-1;
         float extra_elevator = 0;
```

Comment:
i would guess our Qasist bug is the extra elevator, it should be 
if (!is_zero(extra_pitch) && in_vtol_mode() )
this is because the ahrs_view-> pitch_sensor is only correct in vtol modes, i guess alternately you could still apply the extra elevator but use ahrs->pitch_sensor when not in a vtol mode

Reply:
fixed